input/elasticapm/internal/modeldecoder/modeldecoderutil/slice.go

```diff
@@ -40,5 +42,9 @@ func Reslice[Slice ~[]model, model any](slice Slice, n int, newFn func() model)
 		}
 	}
 
+	// slice has enough capacity to hold n elements and there is no need to grow the slice.
```

Comment:
```suggestion
```
This line is misleading as the return also handles the case of growing the slice.